zebra-consensus/src/error.rs

```diff
@@ -41,6 +41,9 @@ pub enum TransactionError {
     #[error("coinbase transaction MUST NOT have the EnableSpendsOrchard flag set")]
     CoinbaseHasEnableSpendsOrchard,
 
+    #[error("coinbase transaction MUST NOT exist in mempool")]
```

Comment:
Here's the coinbase input protocol reference from Bitcoin:
https://developer.bitcoin.org/reference/transactions.html#coinbase-input-the-input-of-the-first-transaction-in-a-block

Using an all-zeroes hash for coinbase input previous transaction IDs, makes it cryptographically impossible for a previous transaction to have that ID. (Even if it's not written that way in the Bitcoin docs.)

```suggestion
    #[error("coinbase inputs MUST NOT exist in mempool")]
```